### PR TITLE
Mesos: 26.0 -> 27.1

### PR DESCRIPTION
**Pull request by kevincox**
###### Things done:
- [x] Tested via `nix.useChroot`.
- [x] Built on platform(s): linux
- [ ] Tested compilation of all pkgs that depend on this change.
- [x] Tested execution of binary products.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### Extra

Fixes # .

cc @ .

---

_Please note, that points are not mandatory._

### CODE REVIEW

**Comment by pSub** [comment]
You haven't changed the hash of the source file.

**Comment by kevincox** [comment]
Oops, forgot to save. Either way I am still building and testing, I just opened the PR so that I could see what Travis thought about it. I have added the WIP to the title and will make a call for reviews when everything is looking good.

**Comment by kevincox** [comment]
Ok, looks good now. Spark was failing on a checksum error of the source file but that really sounds like not my fault. I rebased on master to hope that it was fixed. This version fixes a bug with SSL on mesos that was causing the connection to be instantly closed, I am running this new package in production now!

**Comment by kamilchm** [comment]
@kevincox I started to play with NixOS in case of provision and manage Mesos clusters. What more tools do you use? NixOPS? I was thinking about building immutable images with tools from https://github.com/NixOS/nixpkgs/tree/master/nixos/modules/virtualisation

PS. There's 0.27.2 coming because of this https://issues.apache.org/jira/browse/MESOS-4754

**Comment by kevincox** [comment]
I'm using NixOps. It isn't super pretty right now but if you want we can talk about how I have it set up offline. 

And yeah, we may as well wait for .2 then. I'll update this PR when it comes out.

File: pkgs/applications/networking/cluster/mesos/default.nix [hunk 1]
```diff
@@ -114,7 +118,7 @@ in stdenv.mkDerivation rec {
     rm -f "$out/lib/${python.libPrefix}"/site-packages/site.py*
     popd
 
-    # optional python dependency for mesos cli 
+    # optional python dependency for mesos clix
```
Comment (benley) [inline pkgs/applications/networking/cluster/mesos/default.nix]: `clix` looks like a typo?

**Comment by kevincox** [comment]
Note that this failure is on master as well: #13703

**Comment by kevincox** [comment]
I've updated this PR and am going to recommend merging this even before 27.2 comes out because there are a number of critical issues that I have fixed.
- mesos-slave segfaults on start because of trying to detect systemd.
- Starting commands using the shell fails because `/bin/sh` doesn't exist.

**Comment by kamilchm** [comment]
:+1: for 0.27.1 with your fixes

**Comment by joachifm** [comment]
Travis failure is unrelated, merging

**Comment by cstrahan** [comment]
Awesome - thanks a ton!

**Pull request merged**